File: PHP/Faciles/F4-Calculadora.php
<?php

#implementar switch con opciones
#sumar, restar, multiplicar, dividir
#La opcion siempre debe ser un numero, del o contrario se pregunta de nuevo.
#Se realiza la operacion

$operacion=false;
echo "CALCULADORA
    1. Sumar
    2. Restar
    3. Multiplicar
    4. Dividir
    5. Salir\n";
While($operacion != 5 && $operacion != 4 && $operacion != 3 && $operacion != 2 && $operacion != 1) {
echo "Por favor ingrese su operacion: ";
$operacion = trim(fgets(STDIN));
}


switch ($operacion) {
    case "1":
        $num1=null;
        $num2=null;
        while( !is_numeric($num1) || !is_numeric($num2) ){
            
        echo "Ingrese el primer numero: ";
        $num1 = trim(fgets(STDIN));
        echo "Ingrese el segundo numero: ";
        $num2 = trim(fgets(STDIN));
        }
        echo "La suma de los elementos es: " . ($num1 + $num2)."\n";
        break;
    case "2":
        $num1=null;
        $num2=null;
        while( !is_numeric($num1) || !is_numeric($num2) ){
        echo "Ingrese el primer numero: ";
        $num1 = trim(fgets(STDIN));
        echo "Ingrese el segundo numero: ";
        $num2 = trim(fgets(STDIN));
        }
        echo "La resta de los elementos es: " . ($num1 - $num2)."\n";
        break;
    case "3":
        $num1=null;
        $num2=null;
        while( !is_numeric($num1) || !is_numeric($num2) ){
        echo "Ingrese el primer numero: ";
        $num1 = trim(fgets(STDIN));
        echo "Ingrese el segundo numero: ";
        $num2 = trim(fgets(STDIN));
        }
        echo "La multiplicacion de los elementos es: " . ($num1 * $num2)."\n";
        break;
    case "4":
        $num1=null;
        $num2=null;
        while( !is_numeric($num1) || !is_numeric($num2) ){
        echo "Ingrese el primer numero: ";
        $num1 = trim(fgets(STDIN));
        echo "Ingrese el segundo numero: ";
        $num2 = trim(fgets(STDIN));
        }
        while ($num2 == 0) {
            echo "No se puede dividir por 0\n";
            echo "Ingrese el segundo numero: ";
            $num2 = trim(fgets(STDIN));
        }
        echo "La division de los elementos es: " . ($num1 / $num2)."\n";
        break;
    case "5":
        echo "Saliendo...\n";
        break;
    default:
        echo "Opcion no valida";
}
?>

File: C/Faciles/F4.Calculadora.c
// 4. Implementa una calculadora basica.

#include <stdio.h>

int suma(int, int);
int resta(int, int);
int multi(int, int);
float div(float, float);

int main(){
    int op = 0;
    int valido = 0;
    do{

printf("Calculadora\n");
printf("1. Suma\n");
printf("2. Resta\n");
printf("3. Multiplicacion\n");
printf("4. Division\n");
printf("5. Salir\n");
printf("Seleccione una opcion: ");
scanf("%d", &op);
    }while(op <= 0 || op > 5);
switch(op){
    case 1:
        do{
        printf("Introduzca el primer numero: ");
        int num1;
        // Si no se introduce un numero valido
        if( scanf("%d", &num1) != 1 ){
            valido = 1;
            printf("Error: Debe introducir un número válido.\n");
            // Limpiar el buffer de entrada <button class="citation-flag" data-index="5">
            while (getchar() != '\n');
            op = 0; // Asegurarse de que el bucle continúe
            continue;
        }
        printf("Introduzca el segundo numero: ");
        int num2;
        if( scanf("%d", &num2) != 1 ){
            valido = 1;
            printf("Error: Debe introducir un número válido.\n");
            while (getchar() != '\n'); // Limpiar el buffer de entrada <button class="citation-flag" data-index="5">
            op = 0; // Asegurarse de que el bucle continúe
            continue;
        }
        printf("El resultado es: %d\n", suma(num1, num2));
        valido = 1;

        }while(valido == 0);
        break;
    case 2:
        do{
        printf("Introduzca el primer numero: ");
        int num1;
        // Si no se introduce un numero valido
        if( scanf("%d", &num1) != 1 ){
            valido = 1;
            printf("Error: Debe introducir un número válido.\n");
            // Limpiar el buffer de entrada <button class="citation-flag" data-index="5">
            while (getchar() != '\n');
            op = 0; // Asegurarse de que el bucle continúe
            continue;
        }
        printf("Introduzca el segundo numero: ");
        int num2;
        if( scanf("%d", &num2) != 1 ){
            valido = 1;
            printf("Error: Debe introducir un número válido.\n");
            while (getchar() != '\n'); // Limpiar el buffer de entrada <button class="citation-flag" data-index="5">
            op = 0; // Asegurarse de que el bucle continúe
            continue;
        }
        printf("El resultado es: %d\n", suma(num1, num2));
        valido = 1;

        }while(valido == 0);
        break;
    case 3:
        do{
        printf("Introduzca el primer numero: ");
        int num1;
        // Si no se introduce un numero valido
        if( scanf("%d", &num1) != 1 ){
            valido = 1;
            printf("Error: Debe introducir un número válido.\n");
            // Limpiar el buffer de entrada <button class="citation-flag" data-index="5">
            while (getchar() != '\n');
            op = 0; // Asegurarse de que el bucle continúe
            continue;
        }
        printf("Introduzca el segundo numero: ");
        int num2;
        if( scanf("%d", &num2) != 1 ){
            valido = 1;
            printf("Error: Debe introducir un número válido.\n");
            while (getchar() != '\n'); // Limpiar el buffer de entrada <button class="citation-flag" data-index="5">
            op = 0; // Asegurarse de que el bucle continúe
            continue;
        }
        printf("El resultado es: %d\n", multi(num1, num2));
        valido = 1;

        }while(valido == 0);
        break;
    case 4:
        do{
        printf("Introduzca el primer numero: ");
        float num1;
        // Si no se introduce un numero valido
        if( scanf("%f", &num1) != 1 ){
            valido = 1;
            printf("Error: Debe introducir un número válido.\n");
            // Limpiar el buffer de entrada <button class="citation-flag" data-index="5">
            while (getchar() != '\n');
            op = 0; // Asegurarse de que el bucle continúe
            continue;
        }
        printf("Introduzca el segundo numero: ");
        float num2;
        if( scanf("%f", &num2) != 1 ){
            valido = 1;
            printf("Error: Debe introducir un número válido.\n");
            while (getchar() != '\n'); // Limpiar el buffer de entrada <button class="citation-flag" data-index="5">
            op = 0; // Asegurarse de que el bucle continúe
            continue;
        }
        printf("El resultado es: %f\n", div(num1, num2));
        valido = 1;

        }while(valido == 0);
        break;
    case 5:
        printf("Saliendo...\n");
        break;
}
    return 0;
}

// Funciones
int suma(int a, int b){
    return a + b;
}

int resta(int a, int b){
    return a - b;
}
int multi(int a, int b){
    return a * b;
}
float div(float a, float b){
    return a / b;
}
